app/core/scheduler/modes.py

```diff
@@ -1,78 +1,127 @@
 # Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
 # For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
 
-import os
-import signal
-from abc import ABC, abstractmethod
+import functools
+from abc import ABC
+from datetime import timedelta
 from enum import Enum
+from typing import ClassVar, FrozenSet, Iterable, Callable, Optional, NoReturn
+from app.config import config
 
-from astropy.time import Time
-
-from app.config import config_collector
-from app.core.components.collector import Collector
-from app.core.components.optimizer import Optimizer
-from app.core.components.optimizer.dummy import DummyOptimizer
-from app.core.components.selector import Selector
-from app.core.programprovider.ocs import read_ocs_zipfile, OcsProgramProvider
-from app.db.planmanager import PlanManager
-from definitions import ROOT_DIR
+from lucupy.minimodel.observation import ObservationStatus, Observation
 
 
 class SchedulerMode(ABC):
+    """Base Scheduler Mode.
 
-    @abstractmethod
-    def schedule(self, start: Time, end: Time):
-        pass
-
+    Right now the magic method `__str__` is the only thing that inheritance from this.
+    Also good for duck typing.
+    """
     def __str__(self) -> str:
         return self.__class__.__name__
 
 
 class SimulationMode(SchedulerMode):
-    def schedule(self, start: Time, end: Time):
-        ...
+    pass
 
 
 class ValidationMode(SchedulerMode):
```

Comment:
Main changes in the modes are the removal of the `schedule`. I found it weird that the modes were doing the scheduling instead of ... the Scheduler.  So now we just have specific behavior for each mode inside it. This can grow to be more complex in the future and might need structure.